client/packages/common/src/plugins/types.ts

```diff
@@ -11,19 +11,21 @@ export type ComponentPluginData<T> = extractDataType<
   Extract<ComponentPlugin, { type: T }>
 >;
 
-export type PluginComponent<T> = FunctionComponent<{ data: T }>;
+// export type PluginComponent<T> = FunctionComponent<{ data: T }>;
+
+export type PluginModule<DataType> = {
+  default: React.ComponentType<{ data?: DataType | undefined }>;
+};
 
 export interface ComponentPluginBase<T> {
-  Component: PluginComponent<T>;
-  isLoaded: boolean;
+  component: () => Promise<PluginModule<T>>;
   module: string;
   localModule?: string;
   name: string;
 }
 
 export interface ColumnPluginBase<T extends RecordWithId> {
-  column: ColumnDefinition<T>;
-  isLoaded: boolean;
+  column: () => Promise<ColumnDefinition<T>>;
   module: string;
   name: string;
```

Comment:
Would it make sense to rename it to `pluginName`? or is it really the name of the column plugin? If the later is the case maybe also add an option name in `PluginColumns` and `PluginComponents` and fallback to the `PluginConfig.name` if not set?

Reply:
it is the pluginName - thanks, have renamed